woocommerce/woocommerce-ios
# Product Settings: edit the Menu order

Resolve #2122 

## Description
This PR implements the Menu Order setting in Product Settings, part of M2.
This feature presents a text field that can accept positive and negative value. The lower the value of the number, the higher the item will be on the product list.

## Changes
- Encoded `menuOrder` in Product model
- Refactored `IntegerInputFormatter` using `NumberFormatter` and removing all the regex + tests
- Implemented the new `ProductMenuOrderViewController`
- Added the new `MenuOrder` row in `ProductSettingsRows`
- The view model of `TextFieldTableViewCell` now accept an input formatter
- Updated tests to support `menuOrder`

## Testing
1. Open a simple product
2. Open product settings
3. Tap in "Menu Order" cell
4. Try to edit the menu order value. -> Make sure that only integer values are accepted
5. Press back and save the product.
6. Go to this product from wp-admin and make sure the value was saved correctly.


## Screenshots
| Product Settings screen            |  Menu Order screen |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-24 at 14 30 56](https://user-images.githubusercontent.com/495617/80213029-d8188e80-8638-11ea-9e61-5b44b8b44410.png) | ![Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-24 at 14 26 06](https://user-images.githubusercontent.com/495617/80213033-db137f00-8638-11ea-92c5-6db19bcfff81.png)


Update release notes:
